pkg/controllers/reconciler.go

```diff
@@ -189,8 +253,16 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	ds := &childDaemonSets.Items[0]
 
+	if UpgradePlugins {
```

Comment:
if upgrade is meant to be called only one, why check this always?

Reply:
@ozhuraki once the upgrade is done, the check is no longer relevant. Check the suggestion above